libs/utils/env.py

```diff
@@ -379,6 +379,14 @@ def _init_target(self, force = False):
             platform = Platform(model='MT8173')
             self.__modules = ['bl', 'cpufreq']
 
+        # Initialize gem5 simulation as a board
+        elif self.conf['board'].upper() == 'GEM5':
+            platform = devlib.platform.gem5.Gem5SimulationPlatform(name='gem5',
+                       gem5_bin=self.conf['gem5_bin'],
```

Comment:
Agree that we should have a 'gem5' subdict and not 'gem5_*'.

It's unfortunate to have to have a generic "gem5" board as this board is special and violates assumptions that you can make about the other boards (e.g. if you see "pixel" you know it's an arm64 board with topology XYZ etc, but if you see "gem5" it could be anything, you need to examine the config further). 

It would be simpler to instead have a fixed set of gem5 boards like "gem5_myplatform", "gem5_myotherplatform" and require adding their parameters somewhere in the LISA source. However gem5 boards are naturally going to be a) ephemeral experimental HW designs or b) confidential future HW designs and neither of these cases plays nicely with coding simulation parameters into the LISA source. 

So let's keep it this way - I just wanted to share my thinking; we'll need to be aware that there's a new "special" type of board and try to find any assumptions in the code that this violates.